src/Xamarin.CodeAnalysis/RequiresSuperAttributeCodeFixProvider.cs

```diff
@@ -0,0 +1,92 @@
+//
+// RequiresSuperAttributeCodeFixProvider.cs:
+//
+// Authors:
+//   Madeline Zhang  <v-madzh@microsoft.com>
+//
+// Copyright 2020 Xamarin Inc.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+
+namespace RequiresSuperAttribute
+{
+	[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (RequiresSuperAnalyzerCodeFixProvider)), Shared]
+	public class RequiresSuperAnalyzerCodeFixProvider : CodeFixProvider
+	{
+		const string Title = "Add base method call";
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds {
+			get { return ImmutableArray.Create (RequiresSuperAttributeAnalyzer.DiagnosticId); }
+		}
+
+		public sealed override FixAllProvider GetFixAllProvider ()
+		{
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+			var diagnostic = context.Diagnostics.First ();
+			var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+			// Find the method declaration identified by the diagnostic.
+			var methodDeclaration =
+			  root.FindToken (diagnosticSpan.Start).Parent.AncestorsAndSelf ()
+			  .OfType<MethodDeclarationSyntax> ().First ();
+
+			// Register a code action that will invoke the fix.
+			context.RegisterCodeFix (CodeAction.Create (Title, c => FixSuperAsync (context.Document, methodDeclaration, c), equivalenceKey: Title), diagnostic);
+		}
+
+		private async Task<Document> FixSuperAsync (Document document,
+		  MethodDeclarationSyntax methodDeclaration,
+		  CancellationToken cancellationToken)
+		{
+			//Find the node to fix.
+			var semanticModel = await document.GetSemanticModelAsync (cancellationToken);
+			var memberSymbol = semanticModel.GetDeclaredSymbol (methodDeclaration) as IMethodSymbol;
+
+			//Generate the new base.[methodName]([arguments]) block. If the method needs to return the base call,
+			//it generates return base.[methodName]([arguments])
+			// TODO: this might be hacky - (.Add(SyntaxFactory.Tab) adds an extra tab so that it's a tab (or SyntaxFactory.Tab) further than the leading
+			//      trivia of the method body block, but we should consider investigating if this behaves as expected when the IDE is set to spaces vs
+			//      tabs / different numbers of spaces per tab
+			var returnString = memberSymbol.ReturnsVoid ? "" : "return ";
+			var parametersString = string.Join (",", memberSymbol.Parameters.Select ((p) => p.Name).ToArray ()); //creates a string of the method's passed-in arguments separated by commas
```

Comment:
string.Join can take an IEnumerable: https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=netframework-4.8#System_String_Join__1_System_String_System_Collections_Generic_IEnumerable___0__

and select returns a IEnumerable, ToArray is redundant and not needed.

Reply:
string.Join on an array is more efficient than on an IEnumerable:
https://source.dot.net/#System.Private.CoreLib/String.Manipulation.cs,fa961ce440526a4d

Reply:
On a `string[]` it can pre-allocate one string and not have to deal with a stringbuilder:
https://source.dot.net/#System.Private.CoreLib/String.Manipulation.cs,d60e316e7a402c5a

Reply:
@Therzok but the ToArray call make it loop again, right?

Reply:
Define loop again. Yes, it would allocate a new `string[]` and have to copy it over. In this case, I would resort to benchmarks to see which version is faster :)

Reply:
I wouldn't worry about it in a code fix provider, unless you are planning to fix hundreds or thousands at once.

Reply:
Simplest, remove the ToArray since is not needed ;)